### ISSUE TITLE
Reversed array (y[::-1] / jnp.flip / lax.rev) combined with a broadcast scalar returns zeros past element 0 on MPS; correct with MLX_DISABLE_COMPILE=1

### ISSUE THREAD

**Issue by suchethac** [tillahoffmann/jax-mps#232]
## Summary

On MPS, a reversed array (`y[::-1]`, `jnp.flip`, `lax.rev`, or any negative step like `y[::-2]`) combined with a **broadcast scalar** (`* 2.0`, `+ 1.0`, `2.0 * ...`) returns the correct first element and then zeros (or the scalar's identity value) for every other element. No error is raised.

```
y = [1, 2, ..., 8] (float32)
y[::-1] * 2.0   cpu=[16. 14. 12. 10.  8.  6.  4.  2.]   mps=[16.  0.  0.  0.  0.  0.  0.  0.]
y[::-1] + 1.0   cpu=[9. 8. 7. 6. 5. 4. 3. 2.]           mps=[9. 1. 1. 1. 1. 1. 1. 1.]
```

The reversal alone is correct, and a reversal combined with a **same-shape** operand is correct (`y[::-1] * y[::-1]`, `y[::-1] * y`, `y[::-1] * jnp.ones_like(y)`). A positive step is correct (`y[::2] * 2.0`). It happens for float32 and int32, 1-D and 2-D (either axis), at every size I tried (8, 64, 6127), under `jax.jit` and eagerly (`jax.disable_jit()`).

`MLX_DISABLE_COMPILE=1` makes every case correct. `JAX_MPS_NO_OPTIMIZE=1` changes nothing. So it looks like the compiled/fused path for a binary op whose inputs are (negative-stride view, scalar): element 0 is read correctly and the rest reads as if the view were forward-contiguous from its data pointer, i.e. past the end of the buffer.

Any reduction downstream inherits it silently: `jnp.trapezoid(y[::-1], y[::-1])` gives `-7.5` instead of `-31.5` (only the first interval survives).

## Environment

- Apple M4 Pro, macOS 26 (Darwin 25.6.0)
- Python 3.12, `jax==0.10.2`, `jaxlib==0.10.2`, `jax-mps==0.10.10` (PyPI wheel, latest)
- `JAX_ENABLE_X64=0`, `JAX_PLATFORMS=mps,cpu`

## Reproducer

```python
"""JAX_PLATFORMS=mps,cpu python repro.py   (add MLX_DISABLE_COMPILE=1 to see it pass)"""
import os
import jax, jax.numpy as jnp, numpy as np

cpu, mps = jax.devices("cpu")[0], jax.devices("mps")[0]
y = np.arange(1.0, 9.0, dtype=np.float32)  # [1, 2, ..., 8]

cases = {
    "y[::-1] * 2.0": lambda y: y[::-1] * 2.0,
    "y[::-1] + 1.0": lambda y: y[::-1] + 1.0,
    "jnp.flip(y) * 2.0": lambda y: jnp.flip(y) * 2.0,
    "y[::-2] * 2.0": lambda y: y[::-2] * 2.0,
    "y[::-1] * y[::-1]": lambda y: y[::-1] * y[::-1],  # two same-shape operands: fine
    "y[::-1]": lambda y: y[::-1],                      # the reversal alone: fine
    "y[::2] * 2.0": lambda y: y[::2] * 2.0,            # positive step: fine
    "jnp.trapezoid(y[::-1], y[::-1])": lambda y: jnp.trapezoid(y[::-1], y[::-1]),
}

print(f"jax {jax.__version__}  MLX_DISABLE_COMPILE={os.environ.get('MLX_DISABLE_COMPILE', '<unset>')}")
for name, f in cases.items():
    out = {}
    for dev in (cpu, mps):
        with jax.default_device(dev):
            out[dev.platform] = np.asarray(jax.jit(f)(jnp.asarray(y)))
    ok = np.array_equal(out["cpu"], out["mps"])
    print(f"{'ok   ' if ok else 'WRONG'}  {name:<32} cpu={out['cpu']}  mps={out['mps']}")

# Not a jit artifact: the same happens op by op.
with jax.default_device(mps), jax.disable_jit():
    print("eager y[::-1] * 2.0 on mps:", np.asarray(jnp.asarray(y)[::-1] * 2.0))
```

## Output

Default:

```
jax 0.10.2  MLX_DISABLE_COMPILE=<unset>
WRONG  y[::-1] * 2.0                    cpu=[16. 14. 12. 10.  8.  6.  4.  2.]  mps=[16.  0.  0.  0.  0.  0.  0.  0.]
WRONG  y[::-1] + 1.0                    cpu=[9. 8. 7. 6. 5. 4. 3. 2.]  mps=[9. 1. 1. 1. 1. 1. 1. 1.]
WRONG  jnp.flip(y) * 2.0                cpu=[16. 14. 12. 10.  8.  6.  4.  2.]  mps=[16.  0.  0.  0.  0.  0.  0.  0.]
WRONG  y[::-2] * 2.0                    cpu=[16. 12.  8.  4.]  mps=[16.  0.  0.  0.]
ok     y[::-1] * y[::-1]                cpu=[64. 49. 36. 25. 16.  9.  4.  1.]  mps=[64. 49. 36. 25. 16.  9.  4.  1.]
ok     y[::-1]                          cpu=[8. 7. 6. 5. 4. 3. 2. 1.]  mps=[8. 7. 6. 5. 4. 3. 2. 1.]
ok     y[::2] * 2.0                     cpu=[ 2.  6. 10. 14.]  mps=[ 2.  6. 10. 14.]
WRONG  jnp.trapezoid(y[::-1], y[::-1])  cpu=-31.5  mps=-7.5
eager y[::-1] * 2.0 on mps: [16.  0.  0.  0.  0.  0.  0.  0.]
```

With `MLX_DISABLE_COMPILE=1`: every line `ok`, and the eager line prints `[16. 14. 12. 10.  8.  6.  4.  2.]`.

With `JAX_MPS_NO_OPTIMIZE=1`: identical to the default (still WRONG).

## How it surfaced

In an SED-fitting code (tengri), a polar-dust luminosity is a trapezoid over a frequency grid built from an ascending wavelength grid, so it reverses both arrays before integrating: `jnp.trapezoid(l_nu[::-1], nu[::-1])`. On MPS that integral came out as its first interval only, and after a `jnp.maximum(..., 0)` clamp as exactly `0.0`, which propagated into a far-infrared band being 25x too faint with every other band within float32 noise of the CPU result. Every sub-expression evaluated on its own agreed with CPU, because the reversal only goes wrong once it is fused with a scalar operand. I mention it because the failure is silent and shape-preserving, so a parity check against CPU is the only thing that catches it.

Happy to test a fix or a wheel.

**Comment by suchethac** [comment]
A few more shapes, same environment, to narrow the kernel-selection question (`y = [1..8]` float32, jitted, CPU vs MPS):

```
ok     (y * 2.0)[::-1]                        mps=[16. 14. 12. 10.  8.  6.  4.  2.]   # scalar BEFORE the reversal is fine
WRONG  (y * 2.0)[::-1] * 3.0                  mps=[48.  0.  0.  0.  0.  0.  0.  0.]
WRONG  jnp.sum(y[::-1] * 2.0)                 cpu=72.0  mps=16.0
WRONG  y[::-1] * s   (s a traced 0-d arg)     mps=[16.  0.  0.  0.  0.  0.  0.  0.]   # not only literal constants
WRONG  y[::-1] * jnp.full_like(y, 2.0)        mps=[16.  0.  0.  0.  0.  0.  0.  0.]   # a broadcast full array too
ok     y[::-1] * jnp.ones_like(y)                                                     # (multiply by one presumably simplified away)
WRONG  jnp.where(y[::-1] > 4, y[::-1], 0.0)   cpu=[8. 7. 6. 5. 0. 0. 0. 0.]  mps=[8. 0. 0. 0. 0. 0. 0. 0.]
ok     jnp.exp(y[::-1])                       # unary op directly on the reversed view is fine
WRONG  jnp.exp(-y[::-1])                      mps=[3.35e-04 1. 1. 1. 1. 1. 1. 1.]     # neg(rev(y)) reads as zeros past element 0
ok     -jnp.trapezoid(y, -y)                  # no reversal anywhere: fine (this is the workaround I am using)
```

So the trigger is a reversed (negative-stride) view meeting a broadcast/constant operand in the same fused kernel — literal, traced scalar, or `full_like` — and `neg` behaves like one (I assume it lowers as multiply by -1). Unary `exp` and same-shape operands (`rev * y`, `rev * rev`) are fine, and applying the scalar before reversing is fine.

**Comment by MasterSkepticista** [comment]
This seems to be patched in `mlx` : (https://github.com/ml-explore/mlx/pull/3720)

Can you test #233?

**Comment by suchethac** [comment]
Confirmed with #233's wheel (0.10.11): every case above is correct under default compile, and the integral that first exposed this matches CPU again. Thanks!
